packages/espressocash_app/lib/features/kyc_sharing/data/kyc_repository.dart

```diff
@@ -0,0 +1,166 @@
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:injectable/injectable.dart' hide Order;
+import 'package:kyc_client_dart/kyc_client_dart.dart';
+
+import '../../../utils/errors.dart';
+import '../../accounts/auth_scope.dart';
+import '../../accounts/models/ec_wallet.dart';
+
+@Singleton(scope: authScope)
+class KycRepository extends ChangeNotifier {
+  KycRepository(this._ecWallet);
+
+  final ECWallet _ecWallet;
+  late KycUserClient _kycUserClient;
+
+  Future<void>? _clientInitialization;
+
+  Future<void> _init() => _clientInitialization ??= Future(() async {
+        try {
+          _kycUserClient = _createClient();
+          await _kycUserClient.init(
+            walletAddress: _ecWallet.publicKey.toString(),
+          );
+        } on Exception catch (exception) {
+          _clientInitialization = null;
+          reportError(exception);
```

Comment:
Shouldn't it rethrow error? Otherwise, if initialization failed, operation will still try to run.

Reply:
Actually yeah, better to rethrow it, will add this